### PR TITLE
Add schema visitor.

### PR DESCRIPTION
**Pull request by liurenjie1024**
This is the second part of introducing kudo, for a more complete pr please refer to https://github.com/NVIDIA/spark-rapids-jni/pull/2532. 

In this pr we introduced a post order visitor for schema and host column vectors.

### CODE REVIEW

File: src/main/java/com/nvidia/spark/rapids/jni/schema/SchemaVisitor.java
Comment (jlowe) [inline src/main/java/com/nvidia/spark/rapids/jni/schema/SchemaVisitor.java:47]: This just says "Visit with ..." which begs the question of visit _what_?  Is this visitTopSchema?

File: src/main/java/com/nvidia/spark/rapids/jni/schema/HostColumnsVisitor.java
Comment (jlowe) [inline src/main/java/com/nvidia/spark/rapids/jni/schema/HostColumnsVisitor.java:63]: It's a bit odd and asymmetric that lists get a previsit before children but structs do not.  Curious why this is needed?

**Review by jlowe — APPROVED** [review]
Overall lgtm.  Minor documentation comment and a question on asymmetric previsits.

**Review by GaryShen2008 — APPROVED** [review]
Approved the comment fixing after Jason's approval.

**Pull request merged**
